### PR TITLE
Remove unused config and functions

### PR DESCRIPTION
**Pull request by YangKeao**

### CODE REVIEW

File: scheduler/pkg/mock/mockoption/mockoption.go [hunk 1]
```diff
@@ -30,41 +30,33 @@ const (
...
-	EnableRemoveExtraReplica     bool
-	DisableRemoveDownReplica     bool
-	DisableReplaceOfflineReplica bool
-	DisableMakeUpReplica         bool
-	DisableRemoveExtraReplica    bool
-	LeaderScheduleStrategy       string
+	RegionScheduleLimit         uint64
+	LeaderScheduleLimit         uint64
+	ReplicaScheduleLimit        uint64
+	MergeScheduleLimit          uint64
```
Comment (Connor1996) [inline scheduler/pkg/mock/mockoption/mockoption.go]: merge related can be removed

File: scheduler/pkg/mock/mockoption/mockoption.go [hunk 2]
```diff
@@ -30,41 +30,33 @@ const (
...
+	MaxSnapshotCount            uint64
+	MaxPendingPeerCount         uint64
+	MaxMergeRegionSize          uint64
+	MaxMergeRegionKeys          uint64
+	SchedulerMaxWaitingOperator uint64
+	EnableOneWayMerge           bool
```
Comment (Connor1996) [inline scheduler/pkg/mock/mockoption/mockoption.go]: all `EnableXXX` can be removed

File: scheduler/pkg/mock/mockoption/mockoption.go [hunk 3]
```diff
@@ -30,41 +30,33 @@ const (
...
+	EnableCrossTableMerge       bool
+	MaxStoreDownTime            time.Duration
+	MaxReplicas                 int
+	TolerantSizeRatio           float64
```
Comment (Connor1996) [inline scheduler/pkg/mock/mockoption/mockoption.go]: can be removed

File: scheduler/pkg/mock/mockoption/mockoption.go [hunk 4]
```diff
@@ -30,41 +30,33 @@ const (
...
-	LeaderScheduleStrategy       string
+	RegionScheduleLimit         uint64
+	LeaderScheduleLimit         uint64
+	ReplicaScheduleLimit        uint64
+	MergeScheduleLimit          uint64
+	MaxSnapshotCount            uint64
+	MaxPendingPeerCount         uint64
+	MaxMergeRegionSize          uint64
+	MaxMergeRegionKeys          uint64
+	SchedulerMaxWaitingOperator uint64
```
Comment (Connor1996) [inline scheduler/pkg/mock/mockoption/mockoption.go]: can be removed

File: scheduler/pkg/mock/mockoption/mockoption.go [hunk 5]
```diff
@@ -30,41 +30,33 @@ const (
...
+	EnableOneWayMerge           bool
+	EnableCrossTableMerge       bool
+	MaxStoreDownTime            time.Duration
+	MaxReplicas                 int
+	TolerantSizeRatio           float64
+	EnableRemoveDownReplica     bool
+	EnableReplaceOfflineReplica bool
+	EnableMakeUpReplica         bool
+	EnableRemoveExtraReplica    bool
+	LeaderScheduleStrategy      string
```
Comment (Connor1996) [inline scheduler/pkg/mock/mockoption/mockoption.go]: can be removed

File: scheduler/server/schedule/checker/replica_checker.go [hunk 6]
```diff
@@ -72,30 +72,6 @@ func (r *ReplicaChecker) Check(region *core.RegionInfo) *operator.Operator {
 		return op
 	}
 
-	if len(region.GetPeers()) < r.cluster.GetMaxReplicas() && r.cluster.IsMakeUpReplicaEnabled() {
```
Comment (Connor1996) [inline scheduler/server/schedule/checker/replica_checker.go]: Please keep these

Comment (YangKeao) [reply scheduler/server/schedule/checker/replica_checker.go]: OK!

**Pull request merged**